tests/foreman/api/test_gpgkey.py

```diff
@@ -28,3 +41,213 @@ def test_positive_search_in_org(self):
         gpg_keys = gpg_key.search({'organization'})
         self.assertEqual(len(gpg_keys), 1)
         self.assertEqual(gpg_key.id, gpg_keys[0].id)
+
+    @tier1
+    @run_only_on('sat')
+    def test_positive_create_with_name(self):
+        """Create a GPG key with valid name.
+
+        @Assert: A GPG key is created with the given name.
+
+        @Feature: GPGKey
+        """
+        for name in valid_data_list():
+            with self.subTest(name):
+                gpg_key = entities.GPGKey(
+                    organization=self.org, name=name).create()
+                self.assertEqual(name, gpg_key.name)
+
+    @tier1
+    @run_only_on('sat')
+    def test_positive_create_with_content_text(self):
+        """Create a GPG key with valid name and valid gpg key text.
+
+        @Assert: A GPG key is created with the expected content.
+
+        @Feature: GPGKey
+        """
+        gpg_key = entities.GPGKey(
+            organization=self.org, content=self.key_content).create()
+        self.assertEqual(self.key_content, gpg_key.content)
+
+    @tier1
+    @run_only_on('sat')
+    def test_positive_create_with_content_path(self):
+        """Create a GPG key with valid name and valid gpg key location.
+
+        @Assert: A GPG key is created with the expected content.
+
+        @Feature: GPGKey
+        """
+        gpg_key = entities.GPGKey(
+            organization=self.org, content=self.key_path).create()
+        self.assertEqual(self.key_path, gpg_key.content)
```

Comment:
It's API test, not UI. Please point me to a doc where we can pass the path to a file via API.
AFAIK we have different implementations for UI/CLI/API. For CLI, for example, you can't pass the key in any way - only path to the file. 
According to doc API seems to work vice versa - only key is accepted, no ability to pass the file. Anyway, that's just a guess :) Could you please execute your API test and take a look through UI if the key was fetched? That would explain everything, maybe we just don't have this feature documented.